addons/mrp/models/mrp_workorder.py

```diff
@@ -428,10 +432,12 @@ def write(self, values):
                         values['date_planned_finished'] = computed_finished_time
                     elif values.get('date_planned_start'):
                         computed_duration = workorder._calculate_duration_expected(date_planned_start=start_date)
-                        values['duration_expected'] = computed_duration
+                        if computed_duration:
```

Comment:
since duration_expected has a default value of 60, if we computed that the duration is 0 I think we should set it to 0 (so maybe there should be a `if ... == False:`)

side note: the two elif could be combined `elif values.get('date_planned_start') or values.get('date_planned_finished'):`.

side note 2: doing the equivalent to `_onchange_date_planned_finished` and not fixing it in the method would consist of doing:

```diff
- elif values.get('date_planned_start'):
+ elif values.get('date_planned_start') and end_date:
 ...
- elif values.get('date_planned_finished'):
+ elif values.get('date_planned_finished') and start_date:
```